doc/devdocs/common/FilePreviewCommon.md

```diff
@@ -0,0 +1,70 @@
+# [FilePreviewCommon](/src/common/FilePreviewCommon)
+
+This project contains common code used for previewing and displaying files.
+
+## Monaco preview
+
+Monaco preview enables to display developer files. It is based on [Microsoft's Monaco Editor](https://microsoft.github.io/monaco-editor/) which is maintained by the Visual Studio Code team.
+
+This previewer is used for the File Explorer Dev File Previewer, aswell as PowerToys Peek.
+
+### Update Monaco Editor
+
+1. Download Monaco editor with [npm](https://www.npmjs.com/): Enter `npm i monaco-editor` in the command prompt.
+2. Delete everything except the `min` folder (the minimised code) from the downloaded files.
+3. Copy the `min` folder into the `src/common/FilePreviewCommon/monacoSRC` folder of the PowerToys project.
+4. Generate the JSON file as described in the [Generate monaco_languages.json file](#monaco_languagesjson) section.
+
+### Add a new language definition
+
+As an example on how to add a new language definition you can look at the one for [registry files](/src/common/FilePreviewCommon/customLanguages/reg.js).
+
+1. Add the language definition (written with [Monarch](https://microsoft.github.io/monaco-editor/monarch.html)) to the [`monacoSRC`](/src/common/FilePreviewCommon/customLanguages/) folder. The file should be formatted like in the example below. (Please change `idDefinition` to the name of your language.)
+
+```javascript
+export function idDefinition() {
+    return {
+        ...
+    }
+}
+```
+
+2. Add the following line to the [`monacoSpecialLanguages.js`](/src/common/FilePreviewCommon/monacoSpecialLanguages.js) file:
+
+```javascript
+import { idDefinition } from './customLanguages/file.js';
+```
+
+3. In the [`monacoSpecialLanguages.js`](/src/modules/previewpane/MonacoPreviewHandler/monacoSpecialLanguages.js) file add the following line to the `registerAdditionalLanguages` function:
+
+```javascript
+registerAdditionalNewLanguage("id", [".fileExtension"], idDefinition(), monaco)
+```
+
+  * The id can be anything. Recommended is one of the file extensions. For example "php" or "reg".
+4. Copy the existing language definition into the `languageDefinitions` function in the same file. You can find the existing definitions in the following folder: [`/src/common/FilePreviewCommon/monacoSRC/min/vs/basic-languages`](/src/common/FilePreviewCommon/monacoSRC/min/vs/basic-languages).
```

Comment:
path

Reply:
also, why is this needed? Can you pick any language definition? Can we add some simple placeholder lang definition?

Reply:
Yes cou can pick any. Yes we could, but not in this PR.